### ISSUE TITLE
[BUG] regression between 1.0.6 and 1.0.7

### ISSUE THREAD

**Issue by jmbarbier** [compodoc/compodoc#473]
```
Unhandled Rejection at: Promise {
  <rejected> { SyntaxError: Bad identifier as unquoted key at line 1 column 3 of the JSON5 data. Still to read: "[\"path\":\"err\",\"compo"
    at error (/data/devel/myproject/node_modules/json5/lib/json5.js:56:25)
    at identifier (/data/devel/myproject/node_modules/json5/lib/json5.js:112:17)
    at object (/data/devel/myproject/node_modules/json5/lib/json5.js:454:31)
    at value (/data/devel/myproject/node_modules/json5/lib/json5.js:482:20)
    at Object.parse (/data/devel/myproject/node_modules/json5/lib/json5.js:508:18)
    at loopModulesParser (/data/devel/myproject/node_modules/@compodoc/compodoc/dist/index-cli.js:867:48)
    at RouterParserUtil.constructRoutesTree (/data/devel/myproject/node_modules/@compodoc/compodoc/dist/index-cli.js:901:13)
    at Dependencies.getDependencies (/data/devel/myproject/node_modules/@compodoc/compodoc/dist/index-cli.js:4853:45)
    at CliApplication.Application.getDependenciesData (/data/devel/myproject/node_modules/@compodoc/compodoc/dist/index-cli.js:6276:40)
    at /data/devel/myproject/node_modules/@compodoc/compodoc/dist/index-cli.js:6140:23
  message: 'Bad identifier as unquoted key at line 1 column 3 of the JSON5 data. Still to read: "[\\"path\\":\\"err\\",\\"compo"',
  at: 2,
  lineNumber: 1,
  columnNumber: 3 } } reason: { SyntaxError: Bad identifier as unquoted key at line 1 column 3 of the JSON5 data. Still to read: "[\"path\":\"err\",\"compo"
    at error (/data/devel/myproject/node_modules/json5/lib/json5.js:56:25)
    at identifier (/data/devel/myproject/node_modules/json5/lib/json5.js:112:17)
    at object (/data/devel/myproject/node_modules/json5/lib/json5.js:454:31)
    at value (/data/devel/myproject/node_modules/json5/lib/json5.js:482:20)
    at Object.parse (/data/devel/myproject/node_modules/json5/lib/json5.js:508:18)
    at loopModulesParser (/data/devel/myproject/node_modules/@compodoc/compodoc/dist/index-cli.js:867:48)
    at RouterParserUtil.constructRoutesTree (/data/devel/myproject/node_modules/@compodoc/compodoc/dist/index-cli.js:901:13)
    at Dependencies.getDependencies (/data/devel/myproject/node_modules/@compodoc/compodoc/dist/index-cli.js:4853:45)
    at CliApplication.Application.getDependenciesData (/data/devel/myproject/node_modules/@compodoc/compodoc/dist/index-cli.js:6276:40)
    at /data/devel/myproject/node_modules/@compodoc/compodoc/dist/index-cli.js:6140:23
  message: 'Bad identifier as unquoted key at line 1 column 3 of the JSON5 data. Still to read: "[\\"path\\":\\"err\\",\\"compo"',
  at: 2,
  lineNumber: 1,
  columnNumber: 3 }

```

##### **Overview of the issue**

Doc build works with 1.0.6 (without routing), but fails with the above error in 1.0.7. 
This is part of a big project, with lazy loaded routes. No more infos are provided along the error to be able to locate the route that causes the problem.

Commenting / removing all routes contents, doc building still fails. Only by removing all routing (`RouterModule.forRoot(MAIN_ROUTES)` below, the doc builds)

```typescript
import { NgModule } from "@angular/core";
import { RouterModule, Routes } from "@angular/router";
const MAIN_ROUTES: Routes = [ 
// previously lots of lazy loaded routes + few direct routes
];

@NgModule({
  imports: [
    RouterModule.forRoot(MAIN_ROUTES)  // This is enough to fail
  ],
  exports: [RouterModule],
  providers: []
})
export class AppRoutingModule {
}
```

##### **Operating System, Node.js, npm, compodoc version(s)**

OSX 10.13.4 (17E139j)
node 9.3.0
npm 5.0.6
compodoc 1.0.7

##### **Angular configuration, a `package.json` file in the root folder**

```
{
  "name": "myproject",
  "version": "0.1.0",
  "license": "MIT",
  "scripts": {
    "ng": "ng",
    "start": "ng serve",
    "prebuild": "gulp prebuild:dev",
    "build": "ng build",
    "build:dev": "ng build --prod --environment=dev",
    "prebuild:staging": "gulp prebuild:staging",
    "build:staging": "ng build --prod --stats-json --environment=staging",
    "postbuild:staging": "gulp compress",
    "prebuild:prod": "gulp prebuild:staging",
    "build:prod": "ng build --prod --aot --environment=prod",
    "postbuild:prod": "gulp compress",
    "doc:build": "compodoc -p src/tsconfig.app.json --output docs --theme readthedocs --disableGraph --toggleMenuItems all",
    "doc:serve": "compodoc -s --output docs",
    "doc:buildandserve": "compodoc -p src/tsconfig.app.json -s  --output docs --theme readthedocs --disableGraph --toggleMenuItems all",
    "test": "ng test",
    "lint": "ng lint",
    "lintfix": "ng lint --fix",
    "e2e": "ng e2e",
    "pug": "gulp pug",
    "compress": "gulp compress"
  },
  "private": true,
  "dependencies": {
    "@agm/core": "^1.0.0-beta.2",
    "@angular/animations": "^5.2.5",
    "@angular/common": "^5.2.5",
    "@angular/compiler": "^5.2.5",
    "@angular/core": "^5.2.5",
    "@angular/forms": "^5.2.5",
    "@angular/http": "^5.2.5",
    "@angular/platform-browser": "^5.2.5",
    "@angular/platform-browser-dynamic": "^5.2.5",
    "@angular/router": "^5.2.5",
    "@angular/service-worker": "^5.2.5",
    "@ng-bootstrap/ng-bootstrap": "^1.0.0",
    "@solidev/bsadmincomponents": "0.3.5",
    "@solidev/ngdataservice": "0.4.4",
    "bootstrap": "4.0.0",
    "core-js": "^2.5.3",
    "lodash-es": "^4.17.5",
    "moment": "^2.20.1",
    "rxjs": "^5.5.6",
    "socket.io": "^2.0.4",
    "socket.io-client": "^2.0.4",
    "zone.js": "^0.8.20"
  },
  "devDependencies": {
    "@angular/cli": "1.6.8",
    "@angular/compiler-cli": "^5.2.5",
    "@angular/language-service": "^5.2.5",
    "@compodoc/compodoc": "1.0.6",
    "@types/jasmine": "~2.8.6",
    "@types/jasminewd2": "~2.0.3",
    "@types/lodash": "^4.14.102",
    "@types/node": "~9.4.6",
    "@types/socket.io-client": "^1.4.32",
    "codelyzer": "~4.1.0",
    "del": "^3.0.0",
    "gulp": "4.0.0",
    "gulp-awspublish": "^3.3.1",
    "gulp-cli": "^2.0.1",
    "gulp-gzip": "^1.4.2",
    "gulp-pug": "^3.3.0",
    "jasmine-core": "~2.99.1",
    "jasmine-spec-reporter": "~4.2.1",
    "karma": "~2.0.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.4.1",
    "karma-jasmine": "~1.1.1",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.3.0",
    "ts-node": "~4.1.0",
    "tslint": "~5.9.1",
    "typescript": "~2.6"
  }
}
```
##### **Compodoc installed globally or locally ?**

locally

##### **Motivation for or Use Case**

no doc build with 1.0.7

**Comment by davidkarlsen** [comment]
same here:
```
Unhandled Rejection at: Promise {
  <rejected> { SyntaxError: Bad identifier as unquoted key at line 1 column 3 of the JSON5 data. Still to read: "[\"path\":\"\",\"redirect"
    at error (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/json5/lib/json5.js:56:25)
    at identifier (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/json5/lib/json5.js:112:17)
    at object (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/json5/lib/json5.js:454:31)
    at value (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/json5/lib/json5.js:482:20)
    at Object.parse (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/json5/lib/json5.js:508:18)
    at loopModulesParser (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/@compodoc/compodoc/dist/index-cli.js:867:48)
    at RouterParserUtil.constructRoutesTree (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/@compodoc/compodoc/dist/index-cli.js:901:13)
    at Dependencies.getDependencies (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/@compodoc/compodoc/dist/index-cli.js:4853:45)
    at CliApplication.Application.getDependenciesData (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/@compodoc/compodoc/dist/index-cli.js:6276:40)
    at /Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/@compodoc/compodoc/dist/index-cli.js:6140:23
  message: 'Bad identifier as unquoted key at line 1 column 3 of the JSON5 data. Still to read: "[\\"path\\":\\"\\",\\"redirect"',
  at: 2,
  lineNumber: 1,
  columnNumber: 3 } } reason: { SyntaxError: Bad identifier as unquoted key at line 1 column 3 of the JSON5 data. Still to read: "[\"path\":\"\",\"redirect"
    at error (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/json5/lib/json5.js:56:25)
    at identifier (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/json5/lib/json5.js:112:17)
    at object (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/json5/lib/json5.js:454:31)
    at value (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/json5/lib/json5.js:482:20)
    at Object.parse (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/json5/lib/json5.js:508:18)
    at loopModulesParser (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/@compodoc/compodoc/dist/index-cli.js:867:48)
    at RouterParserUtil.constructRoutesTree (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/@compodoc/compodoc/dist/index-cli.js:901:13)
    at Dependencies.getDependencies (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/@compodoc/compodoc/dist/index-cli.js:4853:45)
    at CliApplication.Application.getDependenciesData (/Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/@compodoc/compodoc/dist/index-cli.js:6276:40)
    at /Users/et2448/projects/tac/jfr/tac-jfr-js-client/node_modules/@compodoc/compodoc/dist/index-cli.js:6140:23
  message: 'Bad identifier as unquoted key at line 1 column 3 of the JSON5 data. Still to read: "[\\"path\\":\\"\\",\\"redirect"',
  at: 2,
  lineNumber: 1,
  columnNumber: 3 }
[17:38:04] Sorry, but there was a problem during parsing or generation of the documentation. Please fill an issue on github. (https://github.com/compodoc/compodoc/issues/new)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
make: *** [docs] Error 1
```

**Comment by markwpearce** [comment]
I believe this is the same as #470

**Comment by vogloblinsky** [comment]
@jmbarbier 
Can i have an example of your routes inside MAIN_ROUTES ?

**Comment by jmbarbier** [comment]
Nothing !!! the build fails even with MAIN_ROUTES and all routes empty...

**Comment by vogloblinsky** [comment]
@jmbarbier in the stack trace in your first message, there is an information about a route with `"path":"err"`
Can i have information about this route ?
I cannot reproduce the error with an empty Routes array. I think it is the other one which breaks the stuff.
Thanks

**Comment by jmbarbier** [comment]
I have a minimal reproductible case : https://github.com/jmbarbier/compodoc-routes-bug

blank angular cli repo + 1 component + 1 route

**Comment by warapitiya** [comment]
same issue

**Comment by jmbarbier** [comment]
This problem have been solved somewhere between 1.0.7 and 1.1.2, everything passes without problem with 1.1.2. Closing issue.
